src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -560,8 +560,11 @@ class FormatParser(
   }
 
   def setFieldMissing(rvb: RegionValueBuilder, i: Int) {
-    rvb.setFieldIndex(formatFieldGIndex(i))
-    rvb.setMissing()
+    val idx = formatFieldGIndex(i)
```

Comment:
@tpoterba why are we calling `setFieldMissing` on non-extant fields?

Reply:
the field is actually missing in the VCF, but it could have been pruned. This is encoding that logic.

Reply:
if the field has been pruned, marking it missing is a no-op